### PR TITLE
Track mean relative error across models.

### PR DESCRIPTION
**Pull request by golechwierowicz**
This PR introduces, and integrates a verification module under the --verify flag. The verification module gets model output as an input, model/benchmark reconstruction args, then runs the eager and finally calculates the relative error and compares it against the provided threshold to return the appropriate error code.

Tested with
```
PJRT_DEVICE=CUDA python3 new_xla/benchmarks/experiment_runner.py --dynamo=inductor --xla=None --dynamo=openxla_eval --dynamo=openxla --xla=PJRT --test=train --test=eval --filter='^resnet50$' --suite-name=torchbench --accelerator=cuda --progress-bar --output-dirname=/tmp/output --repeat=1 --print-subprocess --no-resume --verify
```

### CODE REVIEW

**Comment by vanbasten23** [comment]
This is a cool feature. Mind adding some words to the readme file?

File: benchmarks/verifier.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+def verify(output: torch.Tensor,
+           experiment: dict,
+           model_config: dict,
+           experiment_loader: ExperimentLoader,
+           model_loader: ModelLoader,
+           mean_rel_error_tolerance: float,
+           noop: bool = False):
+  """
+    Verify the mean relative error for `output` against eager runtime of the model.
+    If `mean_rel_error_tolerance` is greater than the calculated mean relative error of
```
[the remark this answers is not in the document]
Comment (golechwierowicz) [reply benchmarks/verifier.py:38]: Obviously, thank you!

**Comment by golechwierowicz** [comment]
> This is a cool feature. Mind adding some words to the readme file?

Will do it in the next PR.

**Review by cota — APPROVED** [review]

**Comment by cota** [comment]
Thanks Greg!

File: benchmarks/verifier.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+      return VerificationResult(VerificationCode.PASS, rel_err)
+    else:
+      log_(eager_output, output)
+      return VerificationResult(VerificationCode.FAIL, rel_err)
+  except Exception as e:
+    logger.exception(e)
+    return VerificationResult(VerificationCode.VERIFICATION_FAILED)
+
+
+def log_(lhs, rhs):
```
Comment (frgossen) [inline benchmarks/verifier.py:91]: Should the "private" functions be `_xyz`?

File: benchmarks/verifier.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+def run_(model):
+  inputs = model.example_inputs
+  with model.pick_grad():
+    output = model.model_iter_fn(inputs)
+  return output
+
+
+def apply_eager_config_(experiment):
+  experiment = experiment.copy()
+  experiment['dynamo'] = experiment['xla'] = None
```
Comment (frgossen) [inline benchmarks/verifier.py:109]: Two lines?

**Review by frgossen — APPROVED** [review]
Thank you!

File: benchmarks/verifier.py (hunk 2 above)
Comment (golechwierowicz) [reply benchmarks/verifier.py]: Fixed!

File: benchmarks/verifier.py (hunk 3 above)
Comment (golechwierowicz) [reply benchmarks/verifier.py]: Fixed!

**Pull request merged**